Base/QTCore/qSlicerExtensionsManagerModel.h

```diff
@@ -380,6 +380,33 @@ public slots:
   /// \sa installExtension, scheduleExtensionForUninstall, uninstallScheduledExtensions
   bool downloadAndInstallExtensionByName(const QString& extensionName, bool installDependencies = true, bool waitForCompletion = false);
 
+  /// \brief Download and install an extension from the extensions server.
+  ///
+  /// This function allows the user to download and install an extension
+  /// along with its dependencies from the extensions server. By default,
+  /// it will prompt the user to confirm the installation and restart of
+  /// the application after installation.
+  ///
+  /// When the testing mode is enabled, the function installs the extension
+  /// and its dependencies without any confirmation dialogs. In this case,
+  /// the application is not automatically restarted.
+  ///
+  /// If testing mode is not already enabled, setting you can set the
+  /// \a confirm parameter to false to skip the confirmation dialogs during
+  /// installation..
+  ///
+  /// Similarly, by setting the \a restart parameter to false, you can prevent
+  /// the application from automatically restarting after the installation is
+  /// completed.
+  ///
+  /// \param extensionName The name of the extension to be installed.
+  /// \param confirm Set to false to skip confirmation dialogs (default: true).
```

Comment:
Since there is already an API to easily set `interactive` to `False`, I will remove the `confirm` parameter.